ls1intum/Artemis
# Make sure the latest programming submission commit hash is shown + improved server integration tests

Fix #1960 

### Checklist

### Motivation and Context
As described in #1960, it can happen that the last commit has is outdated or missing in the summary of the exam.

### Description
The final submit REST call now tries to find and replace the latest programming submission to make sure the correct one is shown.

In this PR, we also add additional (realistic and long) server integration tests for the exam participation to make sure the server code is covered and correct.

### Steps for Testing
1. Participate in an exam with programming exercises
2. Make a last second commit or stay offline in Artemis while you do "local IDE" commits so that you do not receive the latest submission 
3. Wait a couple of seconds in the grace period to make sure the last submission arrives on the Artemis server
4. Submit and double check that the last commit hash is shown correctly.
